### PR TITLE
initial CNS volume create/delete recommendations

### PR DESCRIPTION
**Pull request by ekuric**
- this document should be added as part of admin_guide
- it gives overview of results related to create/delete of CNS volumes up to 100 CNS volumes
- currently 100 cns volumes per TSP is supported number, it was possible to create more CNS volumes than 100, but not *supported* at time 

Signed-off-by: Elvir Kuric <elvirkuric@gmail.com>

### CODE REVIEW

**Comment by ekuric** [comment]
@humblec @jeremyeder ptal , @ahardin-rh you too, please check it and move document where necessary

**Comment by ekuric** [comment]
this document contains links to graphs, please get graphs from these location 
https://raw.githubusercontent.com/ekuric/openshift/master/cns/images/create_10_50_100.png
https://raw.githubusercontent.com/ekuric/openshift/master/cns/images/delete_10_50_100.png
to avoid dependency on my local branch

File: admin_guide/cns_recommendations.adoc [hunk 1]
```diff
@@ -0,0 +1,52 @@
+### CNS - Container Native Storage recommendations for OCP
+
+OCP platform offers CNS storage feature which makes easier for OCP users to fulfill their
+storage needs. With CNS solution users and admins are empowered to have together storage and
+application pods running together on same infrastructure and sharing same resources,this leads
+to better resource utilization.
+Follow CNS https://access.redhat.com/documentation/en-us/red_hat_gluster_storage/3.1/html-single/container-native_storage_for_openshift_container_platform_3.4/[Documentation]
+for instructions how to configure CNS as part of OCP cluster.
```
Comment (jeremyeder) [inline admin_guide/cns_recommendations.adoc]: probably want to link to the gluster 3.2 and ocp 3.5 docs?

File: admin_guide/cns_recommendations.adoc [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+on one Trusted Storage Pool ( TSP )
+
+==== CNS volumes create time
+When deploying applications on OCP it is important to estimate how long
+it will take to build environment from storage side in order for an application pod to able to start.
+If application pod requires an PVC (Persistent Volume Claim) then at below graph
+is presented how long it takes for different number of CNS to be created and bound to corresponding PVC.
+All tests were performed on one TSP (Trusted Storage Pool) and using hardware configuration for
+CNS per CNS https://access.redhat.com/documentation/en-us/red_hat_gluster_storage/3.1/html-single/container-native_storage_for_openshift_container_platform_3.4/[Documentation]
+
```
Comment (jeremyeder) [inline admin_guide/cns_recommendations.adoc]: drop url and say "hardware configuration for CNS per documentation" ?

File: admin_guide/cns_recommendations.adoc [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+Dynamic storage provisioning and storage classes were aslo configured and used when provisioning PVC
+
+.Time to create different number of CNS volumes
+image::https://raw.githubusercontent.com/ekuric/openshift/master/cns/images/create_10_50_100.png[Create CNS volumes]
```
Comment (jeremyeder) [inline admin_guide/cns_recommendations.adoc]: have we been including performance graphs anywhere else in our docs? i dont think so -- i wonder if we can just phrase this as "creation time scales linearly up through 100 volumes.  in our tests, each volume takes approximately 6 seconds to be created, allocated and bound to a pod."

File: admin_guide/cns_recommendations.adoc [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+***Theoretically it is possible to create more than 100 CNS volumes in one
+trusted storage pool, but currently it is not supported***
```
Comment (jeremyeder) [inline admin_guide/cns_recommendations.adoc]: Probably best to drop this sentence until CNS actually supports more volumes.

File: admin_guide/cns_recommendations.adoc [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+
+When user deletes PVC claim which is used by an application pod, then that action
+will trigger deletion of CNS volume which was used by PVC.
+Persistent Volume Claims will disappear immediately from ***oc get pvc*** view, however the the time to delete
+and recycle CNS volumes will depend on number of CNS volumes
+
+Below graphs shows these dependencies
+
+.Time to delete different number of CNS volumes
+image::https://raw.githubusercontent.com/ekuric/openshift/master/cns/images/delete_10_50_100.png[Delete CNS volumes]
```
Comment (jeremyeder) [inline admin_guide/cns_recommendations.adoc]: same comment as above, can this be phrased as "deletion time of CNS volumes scales linearly up to 100 volumes"...

**Review by jeremyeder — CHANGES_REQUESTED** [review]

**Comment by ahardin-rh** [comment]
Continuing work in https://github.com/openshift/openshift-docs/pull/4131

@ekuric @jeremyeder Many thanks for this information!

**Pull request closed without merging**